node/engine/chainservice/mock_chain.go

```diff
@@ -37,6 +38,7 @@ func NewMockChain() *MockChain {
 func (mc *MockChain) SubmitTransaction(tx protocols.ChainTransaction) error {
 	eventsToBroadcast := []Event{}
 	mc.txMutex.Lock()
```

Comment:
Is the txMutex needed at this point? `BlockNumMu` seems to protect the same section of code.

Reply:
Removed `txMutex` in favor of `blockNumMu` here: https://github.com/statechannels/go-nitro/pull/1651/commits/be28326341fd26cf4420ee82f7921f4d319c2840